cli/src/alluxio.org/cli/processes/common.go

```diff
@@ -0,0 +1,284 @@
+				} else {
+					log.Logger.Infof("connection to %s closed", host)
+				}
+			}(conn)
+
+			// create and set up a session
+			session, err := conn.NewSession()
+			if err != nil {
+				result := Result{
+					err: err,
+					msg: fmt.Sprintf("cannot create session at %v.", host),
+				}
+				results <- result
+			}
+			defer func(session *ssh.Session) {
+				err := session.Close()
+				if err != nil && err != io.EOF {
+					log.Logger.Infof("session at %s closed, error: %s", host, err)
+				} else {
+					log.Logger.Infof("session at %s closed", host)
+				}
+			}(session)
+
+			session.Stdout = os.Stdout
+			session.Stderr = os.Stderr
+
+			// run command on the session, output errors
+			if err = session.Run(command); err != nil {
+				result := Result{
+					err: err,
+					msg: fmt.Sprintf("run command %v failed at %v", command, host),
+				}
+				results <- result
+			}
+
+			// if no errors, return nil
+			result := Result{
+				err: nil,
+				msg: "",
+			}
+			results <- result
+		}()
+	}
+
+	go func() {
+		wg.Wait()
+		close(results)
+	}()
+
+	return errorHandler(command, hosts, results)
+}
+
+func prepareCommand() (*ssh.ClientConfig, int, error) {
+	// get the current user
+	cu, err := user.Current()
+	if err != nil {
+		return &ssh.ClientConfig{}, -1,
+			stacktrace.Propagate(err, "cannot find current user")
+	}
+	cuName := cu.Username
+	log.Logger.Debugf("current user: %v", cuName)
+
+	// get public key
+	signer, err := getSigner()
+	if err != nil {
+		return &ssh.ClientConfig{}, -1,
+			stacktrace.Propagate(err, "cannot get private key")
+	}
+
+	// set client config with current user and signer
+	config := &ssh.ClientConfig{
+		User: cuName,
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(signer),
+		},
+		Timeout:         5 * time.Second,
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+
+	// find default ssh port
+	port, err := net.LookupPort("tcp", "ssh")
+	if err != nil {
+		return &ssh.ClientConfig{}, -1,
+			stacktrace.Propagate(err, "get default ssh port failed")
+	}
+	return config, port, nil
+}
+
+func addStartFlags(argument string, cmd *env.StartProcessCommand) string {
+	cliPath := filepath.Join(env.Env.EnvVar.GetString(env.ConfAlluxioHome.EnvVar), "bin", "alluxio")
+	var command []string
+	command = append(command, cliPath, argument)
+	if cmd.AsyncStart {
+		command = append(command, "-a")
+	}
+	if cmd.SkipKillOnStart {
+		command = append(command, "-N")
+	}
+	return strings.Join(command, " ")
+}
+
+func addStopFlags(argument string, cmd *env.StopProcessCommand) string {
+	cliPath := filepath.Join(env.Env.EnvVar.GetString(env.ConfAlluxioHome.EnvVar), "bin", "alluxio")
+	var command []string
+	command = append(command, cliPath, argument)
+	if cmd.SoftKill {
+		command = append(command, "-s")
+	}
+	return strings.Join(command, " ")
+}
+
+func getHostnames(mode string) ([]string, error) {
+	if mode != "master" && mode != "worker" && mode != "all" {
```

Comment:
`mode` could be an enum string, but i'm okay with leaving it as a string as long as the 3 possible values are set as string constants (ex. `ModeMaster = "master") and the function callers also use the same string constant.